Start a 'conventions' section in the Appendices (#3350)
matrix-org/matrix-doc: content/appendices.md

    }
}
```

## Conventions for Matrix APIs

> This section is intended primarily to guide those adding new APIs to the Matrix
> protocol. It sets out guidelines which API designers should seek to follow when
> adding new APIs. This is important to maintain consistency with the Matrix
> protocol, and thus improve developer experience.

Reviewer: The original text feels like it says "API" too many times, but my suggestion isn't much better: 
```suggestion
This section is intended primarily to guide API designers when adding to Matrix,
setting guidelines to follow for how those APIs should work. This is important to 
maintain consistency with the Matrix protocol, and thus improve developer 
experience.
```

either is fine

> ### HTTP Endpoint and JSON property naming

Reviewer: We're not calling it a JSON Property, so it shouldn't be an HTTP Endpoint, I think.
```suggestion
### HTTP endpoint and JSON property naming
```
